src/prelude.rs

```diff
@@ -254,11 +254,30 @@ impl RustRuleContext<'_, '_> {
         self.exec_state.base_values().unwrap::<T>(x)
     }
 
+    /// Convert from an egglog value to reference of Rust container type.
+    /// requires mutable access to execution state
+    ///
+    /// # warnning
+    /// return type contains a read guard, don't leak this guard or will get stuck when `rebuild`
+    pub fn value_to_container<T: ContainerValue>(
+        &mut self,
+        x: Value,
+    ) -> Option<impl Deref<Target = T>> {
+        self.exec_state.container_values().get_val::<T>(x)
+    }
+
     /// Convert from a Rust type to an egglog value.
     pub fn base_to_value<T: BaseValue>(&self, x: T) -> Value {
         self.exec_state.base_values().get::<T>(x)
     }
 
+    /// Convert from a Rust container type to an egglog value.
+    pub fn container_to_value<T: ContainerValue>(&mut self, x: T) -> Value {
+        self.exec_state
+            .container_values()
```

Comment:
If we are adding this here, I would think it would also make sense to add it to the top level API like `value_to_container` is?